vllm/compilation/backends.py

```diff
@@ -58,13 +58,12 @@ def make_compiler(compilation_config: CompilationConfig) -> CompilerInterface:
         else:
             logger.debug("Using InductorAdaptor")
             return InductorAdaptor()
-    else:
-        assert compilation_config.backend == "eager", (
-            "Custom backends not supported with CompilationMode.VLLM_COMPILE"
-        )
-
+    elif compilation_config.backend == "eager":
         logger.debug("Using EagerAdaptor")
         return EagerAdaptor()
+    else:
+        logger.debug("Using custom backend: %s", compilation_config.backend)
+        return resolve_obj_by_qualname(compilation_config.backend)()
```

Comment:
Can we check the interface here?
```suggestion
        compiler = resolve_obj_by_qualname(compilation_config.backend)()
        assert isinstance(compiler, CompilerInterface)
        return compiler
```